src/libraries/System.Net.Security/src/System/Net/Security/SslStream.cs

```diff
@@ -748,6 +762,27 @@ public override int ReadByte()
             return bytesRead == 1 ? oneByte[0] : -1;
         }
 
+        public override unsafe int Read(Span<byte> buffer)
+        {
+            ThrowIfExceptionalOrNotAuthenticated();
+
+            fixed (byte* ptr = &MemoryMarshal.GetReference(buffer))
+            {
+                var memoryManager = RentPointerMemoryManager(ref _readPointerMemoryManager, ptr, buffer.Length);
+
+                try
+                {
+                    ValueTask<int> vt = ReadAsyncInternal<SyncReadWriteAdapter>(memoryManager.Memory, default(CancellationToken));
+                    Debug.Assert(vt.IsCompleted, "Sync operation must have completed synchronously");
+                    return vt.GetAwaiter().GetResult();
+                }
+                finally
+                {
+                    _readPointerMemoryManager = memoryManager;
```

Comment:
I suggest having a ReturnPointerMemoryManager that first does Reset(null, 0) before storing it back into the field.